pandas/tests/reshape/test_pivot.py

```diff
@@ -2486,15 +2486,49 @@ def test_crosstab_with_numpy_size(self):
         tm.assert_frame_equal(result, expected)
 
     def test_crosstab_dup_index_names(self):
-        # GH 13279
-        s = pd.Series(range(3), name="foo")
+        # We test that duplicated column names do not produce issues
+        # GH Issue: #22529, GH#13279
 
-        result = pd.crosstab(s, s)
-        expected_index = pd.Index(range(3), name="foo")
-        expected = pd.DataFrame(
-            np.eye(3, dtype=np.int64), index=expected_index, columns=expected_index
+        # Duplicate name shared between rows and columns
+        s1 = pd.Series(range(3), name="foo")
+        s2 = s1 + 1
+        expected = pd.crosstab(s1, s2.rename("bar"))
```

Comment:
can you do any renames on this line, so that you can use tm.assert_frame_equal(result, expected)